### PR TITLE
fix(contract): refactor aToken require

### PR DESCRIPTION
**Pull request by PierrickGT**
Issue: https://github.com/code-423n4/2022-04-pooltogether-findings/issues/4

### CODE REVIEW

File: contracts/AaveV3YieldSource.sol [hunk 1]
```diff
@@ -334,7 +334,7 @@ contract AaveV3YieldSource is ERC20, IYieldSource, Manageable, ReentrancyGuard {
     address _to,
     uint256 _amount
   ) external onlyManagerOrOwner {
-    require(address(_token) != address(aToken), "AaveV3YS/forbid-aToken-transfer");
+    _requireNotAToken(address(_token), "AaveV3YS/forbid-aToken-transfer");
```
Comment (asselstine) [inline contracts/AaveV3YieldSource.sol]: Why is this message different than the others?

**Review by asselstine — CHANGES_REQUESTED** [review]

File: contracts/AaveV3YieldSource.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/AaveV3YieldSource.sol]: Cause for this one we inform the caller that he can't `transfer` out aTokens from the yield source.
For the other ones, we inform the call that he can't update the `allowance` of aTokens.

Comment (asselstine) [reply contracts/AaveV3YieldSource.sol]: It's not really explaining why the call failed though is it? An error message saying "transfer failed" when the transfer fails isn't very helpful

The *condition* for the failure is the same for all of them: you cannot change the aToken.

Comment (PierrickGT) [reply contracts/AaveV3YieldSource.sol]: Fixed in this commit: https://github.com/pooltogether/aave-v3-yield-source/pull/3/commits/4f79bbbcac2411c9617e1d4c9d6bdb3e929460d9

**Review by asselstine — APPROVED** [review]

**Pull request merged**